model/jpa/src/main/resources/META-INF/jpa-changelog-4.7.0.xml

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!--
+  ~ Copyright 2017 Red Hat, Inc. and/or its affiliates
+  ~ and other contributors as indicated by the @author tags.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet author="keycloak" id="4.7.0-offline-sessions">
+        <createIndex tableName="OFFLINE_USER_SESSION" indexName="IDX_OFFLINE_USS_LSR">
+            <column name="LAST_SESSION_REFRESH" type="INT"/>
```

Comment:
If the offline sessions are refreshed too often, frequent updates to this index can cause performance degradation.
This should not be the case with offline sessions so we should be good here, but please doublecheck .

Reply:
Good point. lastSessionRefresh can be updated quite often, so index on this field is not super-ideal :( I am thinking about changing lastSessionRefresh with the "started" field, which should vary more than lastSessionRefresh and it's not updated at all. It will be probably useful for the future if we want to avoid "blocking" the need for preload offline sessions entirely at startup.